### PR TITLE
trenchbroom, ericw-tools: init

### PR DESCRIPTION
**Pull request by astro**
###### Motivation for this change

Quake map editor + tools

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/trenchbroom/default.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      desktopName = "${pname} level editor";
+      icon = "trenchbroom";
+      comment = meta.description;
+      categories = "Development";
+      exec = "trenchbroom";
+    })
+  ];
+
+  meta = with lib; {
+    homepage = "http://kristianduske.com/trenchbroom";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/trenchbroom/default.nix:54]:
```suggestion
    homepage = "https://trenchbroom.github.io/";
```

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 138042` at 0b27b376 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>ericw-tools</li>
    <li>trenchbroom</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/misc/ericw-tools/default.nix:20:3:
```
   |
20 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/applications/misc/trenchbroom/default.nix [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+let
+  ver = {
+    year = "2021";
+    number = "1";
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "TrenchBroom";
+  version = with ver; "${year}.${number}";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/trenchbroom/default.nix:16]: You could probably use ``lib.versions.major/minor`` to parse the git tag. Then auto update r-rynatm/nix-update should work.

Comment (astro) [reply pkgs/applications/misc/trenchbroom/default.nix]: Done. Thanks @SuperSandro2000

**Pull request merged**